### PR TITLE
feat(app/native): rework appNavigate so callkit video button does not end the call

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react/features/app/actions.native.ts [hunk 1]
```diff
@@ -74,8 +80,28 @@ export function appNavigate(uri?: string, options: IReloadNowOptions = {}) {
...
         if (room) {
+
+            // We need to check for the conference
+            // so the authentication process does not get blocked.
+            if (room === currentRoom && conference) {
+
+                // If participant is not yet inside the conference,
+                // we need to check if the location is the same with the previous one.
+                if (!getIsConferenceJoined(getState())) {
+                    const pastLocationURL = conference.getConnection()[JITSI_CONNECTION_URL_KEY];
```
Comment (saghul) [inline react/features/app/actions.native.ts:95]: It's more like "current".

**Review by saghul — CHANGES_REQUESTED** [review]

**Review by saghul — COMMENTED** [review]
A couple of minor nits, LGTM otherwise!

File: react/features/app/actions.native.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
 import { setRoom } from '../base/conference/actions';
+import {
+    getConferenceState,
+    getIsConferenceJoined
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/app/actions.native.ts]: Removed.

File: react/features/app/actions.native.ts [hunk 3]
```diff
@@ -72,11 +78,23 @@ export function appNavigate(uri?: string, options: IReloadNowOptions = {}) {
...
         if (room) {
-            navigateRoot(screen.connecting);
+            if (conference) {
+
+                // We need to check if the location is the same with the previous one.
+                const currentLocationURL = conference?.getConnection()[JITSI_CONNECTION_URL_KEY];
+                const { hostname: currentHostName, pathname: currentPathName } = currentLocationURL;
+
+                if (currentHostName === hostname  && currentPathName === pathname) {
+                    return;
```
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/app/actions.native.ts:92]: Updated.

**Review by saghul — APPROVED** [review]

**Pull request merged**